extension/lib/tools.js

```diff
@@ -0,0 +1,47 @@
+class Tools {
+  /**
+   * @property {Object} path.storage.device
+   * @param {Object} obj
+   * @param {String} path
+   * @return {Boolean}
+   * @public
+   */
+  static propertyExists (obj, path) {
+    if (!obj) return false
+    if (obj && !path) return true
+
+    const props = path.split('.')
+    let currentObject = obj
+
+    for (let i = 0; i < props.length; ++i) {
+      currentObject = currentObject[props[i]]
+      if (!currentObject) return false
+    }
+
+    return true
+  }
+
+  /**
+   * Checks if the given parameter is an object
+   *
+   * @param {Object} obj
```

Comment:
JsDoc type is wrong. Stating it to be an object doesn't make any sense, if checking for it to be one or not.